applications/fishing-map/src/features/map/Map.tsx

```diff
@@ -25,54 +29,54 @@ const mapOptions = {
   customAttribution: 'Global Fishing Watch 2020',
 }
 
-const Map = (): React.ReactElement => {
-  const mapRef = useMapboxRef()
-  const { viewport, onViewportChange, setMapCoordinates } = useViewport()
-  const { latitude, longitude, zoom } = viewport
-  const { generatorsConfig, globalConfig } = useGeneratorsConnect()
-  const [bounds, setBounds] = useState<MiniglobeBounds | undefined>()
-
-  const onZoomInClick = useCallback(() => {
-    setMapCoordinates({ latitude, longitude, zoom: zoom + 1 })
-  }, [latitude, longitude, setMapCoordinates, zoom])
-  const onZoomOutClick = useCallback(() => {
-    setMapCoordinates({ latitude, longitude, zoom: Math.max(1, zoom - 1) })
-  }, [latitude, longitude, setMapCoordinates, zoom])
-
-  const setMapBounds = useCallback(() => {
-    const mapboxRef = mapRef?.current?.getMap()
-    if (mapboxRef) {
-      const rawBounds = mapboxRef.getBounds()
-      if (rawBounds) {
-        setBounds({
-          north: rawBounds.getNorth() as number,
-          south: rawBounds.getSouth() as number,
-          west: rawBounds.getWest() as number,
-          east: rawBounds.getEast() as number,
-        })
-      }
-    }
-  }, [mapRef])
+const Map = memo(
+  ({ style, onMapClick, onMapHover, children }: any): React.ReactElement => {
+    console.log(Math.random())
+    const mapRef = useMapboxRef()
+    const { viewport, onViewportChange } = useViewport()
+    // TODO: Abstract this away
+    const token = GFWAPI.getToken()
+    const transformRequest: (...args: any[]) => MapRequest = useCallback(
+      (url: string, resourceType: string) => {
+        const response: MapRequest = { url }
+        if (resourceType === 'Tile' && url.includes('globalfishingwatch')) {
+          response.headers = {
+            Authorization: 'Bearer ' + token,
+          }
+        }
+        return response
+      },
+      [token]
+    )
+    const { setMapBounds } = useMapBounds()
```

Comment:
Good catch, thanks. 
In the end I went for having the bounds in the map.slice because I like not having to pass down props.
Also removed the `onLoad` callback in favor of simply setting the bounds when the Map component mounts. (`onLoad` was not appropriate because it's only emitted when all tiles have been loaded, which we don't need to wait for, and it's emitted *every time* a tile loads, which we don't need either)